### PR TITLE
fix(spec-renderer): correct path for the id to fix scrolling (TDX-2877)

### PR DESCRIPTION
**Pull request by davidma415**
# Summary
Swagger was doing some weird escaping on the path, so this replaces the space with an underscore instead (which is what the ID actually is being rendered with)

## PR Checklist

* [ ] **Naming & Structure:** the files and package structure use the conventions outlined in the [Creating a Package docs](https://github.com/Kong/public-ui-components/blob/main/docs/creating-a-package.md).
* [ ] **Tests pass:** check the output of all package unit and/or component tests.
  * [ ] If this PR is the result of a bug, test coverage was added accordingly.
* [ ] **Functional:** all changes do not break existing APIs, but if so, a BREAKING CHANGE commit is in place to bump the major version.
* [ ] **Conventional Commits** all commits follow the conventional commit standards [outlined in the main README](https://github.com/Kong/public-ui-components#committing-changes).
* [ ] **Docs:** includes a technically accurate README, and the docs have been updated accordingly based on the changes in this PR.

### CODE REVIEW

File: packages/portal/swagger-ui-web-component/src/utils.js [hunk 1]
```diff
@@ -1,18 +1,22 @@
...
+    'operations',
+    operation.tag ? operation.tag : 'default',
+    operation.operationId,
+  ]
+}
 
-export const operationToSwaggerThingId = (operation) => escapeSwaggerThing(
-  operationToSwaggerThingArray(operation).join('-'),
-)
+export const operationToSwaggerThingId = (operation) => {
```
Comment (adamdehaven) [inline packages/portal/swagger-ui-web-component/src/utils.js:18]: suggestion: is it possible to provide a type interface for `operation`?

**Review by andrewwylde — CHANGES_REQUESTED** [review]
mostly nits but one change requested. Thanks for the work!

**Comment by davidma415** [comment]
@andrewwylde @adamdehaven so I was trying to refactor `utils.js` to be a TypeScript file, but I was missing the import of `helpers` from `swagger-client`. Once I added that back in, I was getting build errors since `helpers` isn't typed - hope we're okay with leaving it for now?

**Review by adamdehaven — DISMISSED** [review]

**Review by andrewwylde — APPROVED** [review]

**Review by andrewwylde — APPROVED** [review]
Gr8, m8

**Review by adamdehaven — APPROVED** [review]
🚢

File: packages/portal/spec-renderer/src/components/SpecDetails.vue [hunk 2]
```diff
@@ -80,8 +80,15 @@ const showAndScrollToOperation = (): void => {
     return
   }
 
-  swaggerRef.value.showOperation(props.activeOperation)
-  swaggerRef.value.scrollToOperation(props.activeOperation)
+  const activeOperation = { ...props.activeOperation }
+
+  if (activeOperation && activeOperation.tag) {
+    const newStr: string = decodeURIComponent(activeOperation.tag.trim().replaceAll(/\s/g, '%20'))
```
Comment (andrewwylde) [inline packages/portal/spec-renderer/src/components/SpecDetails.vue:86]: can you just add a note that this is the same logic from the other repo? I know they're tightly coupled.

**Pull request merged**